src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -80,36 +147,268 @@ public PolicyEntity updatePolicy(@PathParam("domainid") Long domainid, @PathPara
     public PolicyEntity deletePolicy(@PathParam("domainid") Long domainid, @PathParam("policyid") Long policyid,
             PolicyEntity entity) {
 
-        logger.info("Deleting Policies Entity ...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Deleting for Policies Entity Id...:" + policyid);
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
+
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+
+                CriteriaQuery<PolicyEntity> q = cb.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = q.from(PolicyEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("domain").get("Id"), domainid)),
+                        cb.equal(r.get("id"), policyid));
+
+                List<PolicyEntity> result = DomainApis.this.em.createQuery(q).getResultList();
+                if (result.isEmpty() == false) {
+                    em.remove(result.get(0));
+                    return new PolicyEntity();
+                }
+                else {
+                    return null;
+                }
+            }
+        });
     }
 
     /**
      * Lists the Policy Id's for a given domain
      *
      * @return - Policy Id's
      */
-    @Path("/{domainid}")
+    @Path("/{domainid}/policies")
     @GET
     public List<String> getPolicyIds(@PathParam("domainid") Long domainid) {
 
-        logger.info("Listing Policy Ids'");
-	    //TODO Sudhir: Add db calls here
-		return null;
+        logger.info("Listing Policy Ids'for domain Id ...:" + domainid);
+
+
+        return this.txControl.supports(new Callable<List<String>>() {
+
+            @Override
+            public List<String> call() throws Exception {
+                CriteriaBuilder cb = em.getCriteriaBuilder();
+                CriteriaQuery<PolicyEntity> q = cb.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = q.from(PolicyEntity.class);
+                q.select(r).where(cb.and(cb.equal(r.get("domain").get("Id"), domainid)));
+                List<String> policy = new ArrayList<String>();
+                List<PolicyEntity> PolicyList = em.createQuery(q).getResultList();
+                if (PolicyList.isEmpty() == false) {
```

Comment:
see other comments, also if nothing is found it should return empty list

Reply:
ok